wagtail/contrib/forms/views.py

```diff
@@ -2,152 +2,303 @@
+
+    def get_filtering(self):
+        """ Return filering as a dict for submissions queryset """
+        self.select_date_form = SelectDateForm(self.request.GET)
+        filter = dict()
+        if self.select_date_form.is_valid():
+            date_from = self.select_date_form.cleaned_data.get('date_from')
+            date_to = self.select_date_form.cleaned_data.get('date_to')
+            # careful: date_to should be increased by 1 day since the submit_time
+            # is a time so it will always be greater
+            if date_to:
+                date_to += datetime.timedelta(days=1)
+            if date_from and date_to:
+                filter = dict(submit_time__range=[date_from, date_to])
+            elif date_from and not date_to:
+                filter = dict(submit_time__gte=date_from)
+            elif not date_from and date_to:
+                filter = dict(submit_time__lte=date_to)
+        return filter
+
+    def get_csv_filename(self):
+        """ Returns the filename for the generated CSV file """
+        return 'export-{}.csv'.format(
+            datetime.datetime.today().strftime('%Y-%m-%d')
+        )
+
+    def get_csv_response(self, context):
+        """ Returns a CSV response """
+        filename = self.get_csv_filename()
+        response = HttpResponse(content_type='text/csv; charset=utf-8')
+        response['Content-Disposition'] = 'attachment;filename={}'.format(filename)
 
         writer = csv.writer(response)
-        writer.writerow(data_headings)
-        for s in submissions:
-            data_row = []
-            form_data = s.get_data()
-            for name, label in data_fields:
-                val = form_data.get(name)
-                if isinstance(val, list):
-                    val = ', '.join(val)
-                data_row.append(smart_str(val))
+        writer.writerow(context['data_headings'])
+        for data_row in context['data_rows']:
             writer.writerow(data_row)
         return response
 
-    paginator, submissions = paginate(request, submissions)
-
-    data_rows = []
-    for s in submissions:
-        form_data = s.get_data()
-        data_row = []
-        for name, label in data_fields:
-            val = form_data.get(name)
-            if isinstance(val, list):
-                val = ', '.join(val)
-            data_row.append(val)
-        data_rows.append({
-            "model_id": s.id,
-            "fields": data_row
+    def render_to_response(self, context, **response_kwargs):
+        if self.is_csv_export:
+            return self.get_csv_response(context)
+        return super().render_to_response(context, **response_kwargs)
+
+    def get_context_data(self, **kwargs):
+        """ Return context for view, handle CSV or normal output """
+        context = super().get_context_data(**kwargs)
+        submissions = context[self.context_object_name].all()  # force queryset into list
```

Comment:
Why force it here and not leave it to lazily evaluate with the `for` loops below?